src/Integration/NewConnectedMode/ConfigurationSerializer.cs

```diff
@@ -96,13 +78,14 @@ internal static string GetConnectionFilePath(string solutionFilePath)
             return Path.Combine(solutionFolder, ".sonarlint", $"{solutionName}.sqconfig");
         }
 
-        private string GetCurrentSolutionFilePath()
+        protected override string GetFullConfigurationFilePath()
         {
             object fullSolutionName;
             // If there isn't an open solution the returned hresult will indicate an error
             // and the returned solution name will be null. We'll just ignore the hresult.
             solution.GetProperty((int)__VSPROPID.VSPROPID_SolutionFileName, out fullSolutionName);
-            return (string)fullSolutionName;
+
+            return GetConnectionFilePath((string)fullSolutionName);
```

Comment:
Can it not be a string? Is it worth having a safe-cast?

Reply:
I don't think it can be anything else, but there's no harm in playing safe. I'll change it.